### ISSUE TITLE
Scalar and vector projection

### ISSUE THREAD

**Issue by Shylie** [endlesstravel/Love2dCS#77]
Could you please implement scalar and vector projection for vectors? See the [scalar projection wikipedia page](https://en.wikipedia.org/wiki/Scalar_projection) and the [vector projection wikipedia page](https://en.wikipedia.org/wiki/Vector_projection) for more information.

**Comment by endlesstravel** [comment]
ok, I will add them. Is this ok :
```C#
            /// <summary>
            /// the scalar projection of a vector a on (or onto) a vector b
            /// </summary>
            public static float ScalarProjection(Vector2 a, Vector2 b)
            {
                return Vector2.Dot(a, b) / b.Length();
            }

            /// <summary>
            /// the vector projection of a vector a on (or onto) a vector b
            /// </summary>
            public static Vector2 VectorProjection(Vector2 a, Vector2 b)
            {
                return ScalarProjection(a, b) * b;
            }

            /// <summary>
            /// the scalar projection of a vector a on (or onto) a vector b
            /// </summary>
            public static float ScalarProjection(Vector3 a, Vector3 b)
            {
                return Vector3.Dot(a, b) / b.Length();
            }

            /// <summary>
            /// the vector projection of a vector a on (or onto) a vector b
            /// </summary>
            public static Vector3 VectorProjection(Vector3 a, Vector3 b)
            {
                return ScalarProjection(a, b) * b;
            }

            /// <summary>
            /// the scalar projection of a vector a on (or onto) a vector b
            /// </summary>
            /// <param name="a"></param>
            /// <param name="b"></param>
            public static float ScalarProjection(Vector4 a, Vector4 b)
            {
                return Vector4.Dot(a, b) / b.Length();
            }

            /// <summary>
            /// the vector projection of a vector a on (or onto) a vector b
            /// </summary>
            public static Vector4 VectorProjection(Vector4 a, Vector4 b)
            {
                return ScalarProjection(a, b) * b;
            }
```

**Comment by Shylie** [comment]
Looks good. Thank you!

**Comment by endlesstravel** [comment]
`Vector2/3/4.ScalarProjection  VectorProjection`  available from [11.0.34](https://www.nuget.org/packages/Love2dCS/11.0.34)

**Comment by endlesstravel** [comment]
closed because it finished .
